tests/server/test_agents_api.py

```diff
@@ -230,3 +232,51 @@ async def test_sorting_validation(client, environment: str, env_with_agents: Non
     for sort, expected_status in sort_status_map.items():
         result = await client.get_agents(environment, sort=sort)
         assert result.code == expected_status
+
+
+@pytest.mark.asyncio
+async def test_agent_process_details(client, environment: str) -> None:
+    env_uuid = uuid.UUID(environment)
+    process_sid = uuid.uuid4()
+    await data.AgentProcess(
+        hostname="localhost-dummy", environment=env_uuid, sid=process_sid, last_seen=datetime.datetime.now()
+    ).insert()
+    id_primary = uuid.uuid4()
+    await data.AgentInstance(id=id_primary, process=process_sid, name="dummy-instance", tid=env_uuid).insert()
+    await data.Agent(
+        environment=env_uuid,
+        name="dummy-agent",
+        id_primary=id_primary,
+        paused=True,
+    ).insert()
+
+    result = await client.get_agent_process_details(environment, process_sid)
```

Comment:
This API call will always include the `state` field with value None, while `report=False`. I am not sure whether this is a problem. But it feels a bit weird to me.